test/OpenTelemetry.Instrumentation.Runtime.Tests/RuntimeMetricsTests.cs

```diff
@@ -37,6 +39,133 @@ public void RuntimeMetricsAreCaptured()
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                 .AddRuntimeInstrumentation()
+                 .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            var jitCompiledSizeMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.jit.il_compiled.size");
+            Assert.True(GetLongSum(jitCompiledSizeMetric) > 0);
+
+            var jitMethodsCompiledCountMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.jit.methods_compiled.count");
+            Assert.True(GetLongSum(jitMethodsCompiledCountMetric) > 0);
+
+            var jitCompilationTimeMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.jit.compilation_time");
+            Assert.True(GetLongSum(jitCompilationTimeMetric) > 0);
+        }
+#endif
+
+#if NETCOREAPP3_1_OR_GREATER
+        [Fact]
+        public void RuntimeMetrics_ThreadingRelatedMetrics()
+        {
+            var exportedItems = new List<Metric>();
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                 .AddRuntimeInstrumentation()
+                 .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            var lockContentionCountMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.monitor.lock_contention.count");
+            Assert.True(GetLongSum(lockContentionCountMetric) >= 0);
+
+            var threadCountMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.thread_pool.threads.count");
+            Assert.True(GetLongSum(threadCountMetric) > 0);
+
+            var completedItemsCountMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.thread_pool.completed_items.count");
+            Assert.True(GetLongSum(completedItemsCountMetric) > 0);
+
+            var queueLengthMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.thread_pool.queue.length");
+            Assert.True(GetLongSum(queueLengthMetric) == 0);
+
+            // Create 10 timers to bump timer.count metrics.
+            int timerCount = 10;
+            TimerCallback timerCallback = _ => { };
+            List<Timer> timers = new List<Timer>();
+            for (int i = 0; i < timerCount; i++)
+            {
+                Timer timer = new Timer(timerCallback, null, 1000, 250);
+                timers.Add(timer);
+            }
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            var timerCountMetric = exportedItems.First(i => i.Name == "process.runtime.dotnet.timer.count");
+            Assert.True(GetLongSum(timerCountMetric) >= timerCount);
+            for (int i = 0; i < timers.Count; i++)
+            {
+                timers[i].Dispose();
+            }
+        }
+#endif
+
+        private static double GetLongSum(Metric metric)
```

Comment:
1. Why do we use `GetLongSum` for non-sum metrics?
2. I think it would be enough just to check for the existence of the metrics in most of the places e.g.

```csharp
var gcCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.assemblies.count");
Assert.NotNull(gcCountMetric);
```

Reply:
+1. For places where we can "cause" the metric value to be something, we can assert there. Eg: if we have 10 timers, we know the time count should be at least 10.

Reply:
Updated to only check for existence.